### ISSUE TITLE
Driveways in construction look like railways

### ISSUE THREAD

**Issue by RicoElectrico** [gravitystorm/openstreetmap-carto#5047]
### Expected behavior
The style of `highway=construction construction=service service=driveway` is different enough from  railways
### Actual behavior
They're visually too similar
### Screenshots with links illustrating the problem
![image](https://github.com/user-attachments/assets/772e3d02-c784-4ea4-81bb-807a6965b7b6)
https://www.openstreetmap.org/#map=19/50.056987/19.086589

**Comment by dch0ph** [comment]
I suspect this is a "won't fix". I do find the construction rendering a bit jarring - it does stand out / "use more ink", whereas "in construction" roads etc. are generally less prominent ("use less ink") then normal roads. But it would be quite challenge to develop a new consistent style for construction.

Realistically in construction minor service roads are not easily confused with railways from the context.

**Comment by Vectorial1024** [comment]
I have seen maps that handles "roads in construction" with dashed road outlines, but when put into osm-carto it might be ambiguous with tunnels since tunnels also have dashed outlines.

Another approach to this could be to somehow make the railway paths have a "thicker" outline, so that railways differentiates from motor-vehicle roads more. Or, make the railway stripes longer to make it obvious those are railways.

**Comment by dch0ph** [comment]
> Another approach to this could be to somehow make the railway paths have a "thicker" outline, so that railways differentiates from motor-vehicle roads more. Or, make the railway stripes longer to make it obvious those are railways.

I don't think this is a viable route - the railway render works well as it stands, so changing it to address a relatively minor issue seems a non-starter.

Personally I feel more viable option to evaluate would be dropping the (thin) casing on construction renders. You could also try reducing the opacity of the white "fill". If I understand the ordering correctly, construction types have low priority and so junctions should be rendered sensibly.

**Comment by imagico** [comment]
Using the `landuse=construction` color instead of the dark gray for construction on white/gray road types would significantly decrease this problem - see also #4731.
